### DISCUSSION TITLE
What is the purpose of  SYSTEM.maxTextureWidth

### DISCUSSION THREAD

**Discussion by cl1069573062** [mistic100/Photo-Sphere-Viewer#651] [General]
l see __createEquirectangularTexture  means
if image width > SYSTEM.maxTextureWidth  will resize image 
but l  only  use  createTexture(img),   don't  resize image ,  and   is  work
 so  What is the purpose of  SYSTEM.maxTextureWidth , and   resize image means

**Comment by mistic100** [comment]
This is is the maximum WebGL texture size supported by the device, generally 16384 on desktop and 4096 on mobiles.  Bigger textures results in a black image.

And because the resize is done with a canvas it also depends of the maximum canvas size found with `SYSTEM.getMaxCanvasWidth()` (this is not a system constant though and found by trial and error)

**Reply by cl1069573062** [reply]
I uploaded an image  size larger than 4096 pixels on my device,  and my device maximum WebGL texture size  is 4096.
l comment out this part of the code don't resize image，But it still works，It's confusing to me

Sample Pictures  8000 * 4000
![下载](https://user-images.githubusercontent.com/15069666/158309007-42a53ea0-7898-4835-bd98-0eec47430bd5.jpeg)

**Reply by mistic100** [reply]
Do you see a console warning saying "THREE.WebGLRenderer: Texture has been resized from" ... ?
This is performed by THREE.js itself https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLTextures.js

Perhaps when I started/forked this project, THREE.js did not handled it, or I did it myself to avoid the warning. I don't remember.

Either way the resizing is required.

**Reply by cl1069573062** [reply]
I understand. Thank you very much。
